src/components/dialogs/upload-case.js

```diff
@@ -6,20 +6,68 @@
  */
 
 import { useDispatch, useSelector } from 'react-redux';
-import { selectFile } from '../../redux/actions';
+import {
+    selectFile,
+    setFormatWithParameters,
+    setTempCaseUuid,
+    setformatInvalidMsgError,
+} from '../../redux/actions';
 import Button from '@mui/material/Button';
 import { FormattedMessage } from 'react-intl';
-import React from 'react';
-
+import React, { useState } from 'react';
+import {
+    deleteCaseByCaseUuid,
+    getCaseImportParameters,
+    getCaseUuidWhenUploadFile,
+} from '../../utils/rest-api';
+import CircularProgress from '@mui/material/CircularProgress';
 export const UploadCase = () => {
     const dispatch = useDispatch();
     const selectedFile = useSelector((state) => state.selectedFile);
-
+    const tempCaseUuid = useSelector((state) => state.tempCaseUuid);
```

Comment:
I understand that we already gave the selectedFile through the store so you did the same with the caseUuid but this isn't a state that should be stored in the redux store. We only need this info for a short amount of time (as we can see by the naming of the variable). Now the question is: do we keep it that way for this PR and refactor it in another or should we try to do it correctly here ? 
It's the same problem with `formatWithParams` and `formatInvalidMsgError`
What is your thought @SlimaneAmar

Reply:
Yes i think this need a refactor in another PR